misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -164,12 +168,22 @@ internal class RealTransacter private constructor(
 
   internal class RealSession(
     val session: org.hibernate.Session,
-    val config: DataSourceConfig
+    val config: DataSourceConfig,
+    val readOnly: Boolean
   ) : Session {
     override val hibernateSession = session
+    init {
+      if (readOnly) {
+        hibernateSession.isDefaultReadOnly = true
+        hibernateSession.hibernateFlushMode = FlushMode.MANUAL
+      }
+    }
 
     @Suppress("UNCHECKED_CAST")
     override fun <T : DbEntity<T>> save(entity: T): Id<T> {
+      if (readOnly) {
```

Comment:
just wanted to double check that this check is only for redundancy and providing a nice error message. I'm hoping the above hibernate settings are properly flipping the readOnly bit on the jdbc connection. Is that true?